pydantic/_internal/_forward_ref.py

```diff
@@ -10,6 +10,7 @@
 
 if TYPE_CHECKING:
     from pydantic import BaseModel
+    from pydantic.dataclasses import PydanticDataclass
```

Comment:
```suggestion
    from pydantic._internal._dataclasses import PydanticDataclass
```
With the existing import, mypy gives the error:
```
pydantic/_internal/_forward_ref.py:13: error: Module "pydantic.dataclasses" does not explicitly export attribute "PydanticDataclass"  [attr-defined]
```
Minor thing, but I think it's right, and I don't see any reason not to do it this way.